backend/app.py

```diff
@@ -153,10 +140,64 @@ def ec2_instances() -> Response:
             body_token = body.get("token")
             # ensure the id and secret exist
             if body_token:
-                json_val = json.dumps(ec2_instances_mock_data)
+                json_val = json.dumps({"Nothing": "nothing"})
                 return Response(json_val, status=200, mimetype='application/json')
             return Response("Token missing", status=400, mimetype='application/text')
         else:
             return Response("Request body missing", status=400, mimetype='application/text')
 
+    @app.route('/api/google_authentication', methods=["POST"])
+    def google_authentication() -> Response:
+        """
+        A method to authenticate with google.
+
+        Returns
+        -------
+        Response
+            an http response
+        """
+
+        check_endpoint = "https://www.googleapis.com/oauth2/v1/tokeninfo?access_token="
+
+        body = request.get_json()
+
+        if body:
+            auth_email = body.get("email")
+            auth_token = body.get("token")
+            access_key = body.get("access_key")
+            secret_key = body.get("secret_key")
+
+            if all((auth_email, auth_token, access_key, secret_key)):
+                try:
+                    r = requests.get(check_endpoint + auth_token)
+                    data = r.json()
+
+                    if data.get("issued_to") and data.get("issued_to") == auth_email:
+                        token = str(uuid.uuid4())
+                        ttl = 128000
```

Comment:
Maybe make the ttl a standard global variable, or put it in config or something

Reply:
I'll make it a ttl a global for now